sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -81,6 +81,13 @@ private TestProxy(string proxyPath, bool debugMode = false)
                 }
             };
 
+            bool.TryParse(Environment.GetEnvironmentVariable("TF_BUILD"), out bool inCI);
+
+            if (inCI)
+            {
+                testProxyProcessInfo.EnvironmentVariables["PROXY_ASSETS_FOLDER"] = Path.Combine(System.IO.Path.GetTempPath(), Guid.NewGuid().ToString());
```

Comment:
Just curious - does this solve the problem by ensuring each proxy instance is working from a unique file path?  Where else is this environment variable referenced?

Reply:
This is exactly correct. The issue is that all the proxy processes are accessing the same `.breadcrumb` file and we're seeing all the weird issues in .NET runs due to that. I didn't originally intend the proxy to be used that way, but I can't argue that we can support it!

Reply:
> Where else is this environment variable referenced?

This is a [proxy specific variable.](https://github.com/Azure/azure-sdk-tools/blob/e02eeee926c971471a434062ca85a78c855f5eed/tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/GitAssetsConfiguration.cs#L95) Your framework doesn't need to care about it at all.